_posts/2017/2017-12-31-avoid-double-negatives.md

```diff
@@ -0,0 +1,62 @@
+---
+title: 'Avoid Double Negatives'
+description: 'if unless else end'
+layout: article
+category: ruby
+image:
+  base: '2017/avoid-double-negatives'
+  alt: 'Two directions sign'
+  source: 'https://unsplash.com/photos/lPQIndZz8Mo'
+  credit: 'Pablo Garcia Saldaña'
+---
+
+Ruby has a bunch of delightful syntactic sugar. One example of this is `unless`, which you can use in place of an `if` with a negative.
+
+However given the flexibility of Ruby's syntax it is easy to make code harder to understand than it needs to be.
+
+## Instead of…
+
+...overcomplicating your conditionals.
+
+```ruby
+# Example 1
+unless something.nil?
+  # do something
+end
+
+# Example 2
+if !something.nil?
+  # do something
+end
+
+# Example 3
+if !!something
+  # do something
+end
+```
+
+
+## Use…
+
+```ruby
+# Instead of Examples 1,2 & 3
+if something
+  # do something
+end
+```
+
+
+## But why?
+
+Ruby’s conditional syntax is ‘truthy’. In Ruby’s case that means `nil` is equivalent to `false`. There is no need to do a negated `#nil?` check on an object as it is redundant as shown in the first two examples.
+
+The syntax of `!!` in the third example converts a value to boolean, but it is unnecessary and likely obscures your intention. If you _really_ want to do a `nil` check, you should use `#nil?` instead.
```

Comment:
What is the intention here?

Maybe start with explaining why you'd want to use `!!`, i.e. why you want a boolean...then it'll be easier to show why just using `#nil?` makes sense.